app/models/concerns/orchestration/ssh_provision.rb

```diff
@@ -80,7 +81,9 @@ def setSSHProvision
     if client.deploy!
       # since we are in a after_commit callback, we need to fetch our host again, and clean up puppet ca on our own
       Host.find(id).built
-      respond_to?(:initialize_puppetca,true) && initialize_puppetca && delAutosign if puppetca?
+      if puppetca? && respond_to?(:initialize_puppetca, true)
```

Comment:
Ah, I assumed we ignore when returning nil (and only false is significant), but you're right, updated